lib/Docs/RST/DoctrineRSTBuilder.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Docs\RST;
+
+use Doctrine\RST\Builder;
+use Override;
+use phpDocumentor\Guides\Nodes\DocumentNode;
+
+final class DoctrineRSTBuilder implements DocumentsBuilder
+{
+    public function __construct(
+        private Builder $builder,
+    ) {
+    }
+
+    #[Override]
+    public function build(string $directory, string $targetDirectory = 'output'): void
+    {
+        // we have to get a fresh builder due to how the RST parser works
+        $this->builder = $this->builder->recreate();
+
+        // build the docs from the files in $docsDir and write them to $outputPath
+        // which is contained inside the $sourceDir
+        $this->builder->build(
+            $directory,
+            $targetDirectory,
+        );
+    }
+
+    /** @return DocumentNode[] */
+    #[Override]
+    public function getDocuments(): array
+    {
+        return $this->builder->getDocuments()->getAll();
```

Comment:
I agree with static analysis… (why) is this OK? Do both classes maybe have a similar interface?

Reply:
I think the question is, should we invest time on this. Phpstan is right, this is not correct. But I would rather remove this whole class than make it work. As we will move away from the old parser. 

This will be a point of no return when I start the cleanup. This was a first attempt to do things step by step. Which failed.

Reply:
Please be aware that I won't approve PRs that do have failing workflows.